### DISCUSSION TITLE
OpenSSL announces final release of OpenSSL 3.2.0

### DISCUSSION THREAD

**Discussion by hlandau** [openssl/openssl#22806] [Announcements]
We are pleased to announce the immediate availability of OpenSSL 3.2.0. OpenSSL 3.2.0 is the first General Availability release of the OpenSSL 3.2 release line, and incorporates a number of new features, including:

- Client-side QUIC support, including support for multiple streams (RFC 9000)
- Certificate compression in TLS (RFC 8879), including support for zlib, zstd
  and Brotli
- Deterministic ECDSA (RFC 6979)
- Support for Ed25519ctx, Ed25519ph and Ed448ph (RFC 8032) in addition to existing support for Ed25519 and Ed448
- AES-GCM-SIV (RFC 8452)
- Argon2 (RFC 9106) and supporting thread pool functionality
- HPKE (RFC 9180)
- The ability to use raw public keys in TLS (RFC 7250)
- TCP Fast Open (RFC 7413) support, where supported by the OS
- Support for provider-based pluggable signature schemes in TLS, enabling third-party post-quantum and other algorithm providers to use those algorithms with TLS
- Support for Brainpool curves in TLS 1.3
- SM4-XTS
- Support for using the Windows system certificate store as a source of trusted root certificates. This is not yet enabled by default and must be activated using an environment variable. This is likely to become enabled by default in a future feature release.

A complete summary of the major new features and significant changes in OpenSSL 3.2 can be found in the [NEWS file]; a more detailed list of changes in OpenSSL 3.2 can be found in [the CHANGES file on GitHub].

Users interested in using the new QUIC functionality are encouraged to read the [README file for QUIC], which provides links to relevant documentation and example code.

OpenSSL 3.2.0 can be [downloaded as a source tarball here] or obtained from our [release tag on GitHub]. Checksums and release signatures may be found on the [Downloads page].

The next feature release after OpenSSL 3.2 will be OpenSSL 3.3, which will be released no later than 30 April 2024. This release is expected to include QUIC server support. The determination of what other features will be shipped in OpenSSL 3.3 is ongoing and will be decided by our recently announced [Release Steering Committee].

We would like to thank all of our users and communities for their continued use and support of OpenSSL. OpenSSL 3.2.0 represents the product of over two years of development work, comprising over 4,000 commits and contributions from over 300 different authors. This release would not be possible without the innumerable bug reports, pull requests, code reviews and feedback we continue to receive from our community.

We would also like to extend our thanks to all of the organisations who have supported the development of OpenSSL 3.2 financially, whether by holding a [support contract] with us or by [sponsoring OpenSSL]. These organisations provide a [sustainable income source] for the project, and continue to enable us to fund consultants to work full time for the OpenSSL project. [Over 60% of commits] in the past year were funded by the OpenSSL project itself, thanks to our support customers and sponsors.

As always, bug reports and issues relating to OpenSSL can be [filed on our issue tracker], and [questions about using OpenSSL 3.2 can be posted on GitHub Discussions.][discussions]

Comments on this release are also welcomed below, or via email to [feedback@openssl.org].

[View blogpost]

---
[Downloads page]: https://www.openssl.org/source/
[NEWS file]: https://github.com/openssl/openssl/blob/openssl-3.2.0/NEWS.md
[Release Steering Committee]: https://www.openssl.org/blog/blog/2023/08/27/steps-forward/
[time-based release schedule]: https://www.openssl.org/blog/blog/2023/08/27/steps-forward/
[the CHANGES file on GitHub]: https://github.com/openssl/openssl/blob/openssl-3.2.0/CHANGES.md
[downloaded as a source tarball here]: https://www.openssl.org/source/openssl-3.2.0.tar.gz
[release tag on GitHub]: https://github.com/openssl/openssl/releases/tag/openssl-3.2.0
[sustainable income source]: https://www.openssl.org/blog/blog/2023/08/08/finances/
[Over 60% of commits]: https://www.openssl.org/blog/blog/2023/07/17/who-writes-openssl/
[README file for QUIC]: https://github.com/openssl/openssl/blob/openssl-3.2/README-QUIC.md
[sponsoring OpenSSL]: https://www.openssl.org/support/acks.html
[support contract]: https://www.openssl.org/support/contracts.html
[filed on our issue tracker]: https://github.com/openssl/openssl/issues
[feedback@openssl.org]: mailto:feedback@openssl.org
[discussions]: https://github.com/openssl/openssl/discussions
[View blogpost]: https://www.openssl.org/blog/blog/2023/11/23/OpenSSL32/

**Comment by friesoft** [comment]
Congratulations on the release! Is it correct that OpenSSL 3.2 is binary compatible with 3.1 so it should be a drop in replacement and just work with Apache httpd, curl, ... without recompiling those

**Reply by t8m** [reply]
Yes, it is backwards binary compatible with 3.1, and 3.0. So if you compile with 3.0 you should be able to run with 3.2. Not the other way around as there can be API additions that will be obviously missing on older versions.

**Reply by hlandau** [reply]
Indeed &mdash; we have a strict API/ABI compatibility policy. Only major releases (i.e. 4.x) will be potentially incompatible. You can find more information in our [versioning policy](https://github.com/openssl/general-policies/blob/master/policies/versioning-policy.md).

When considering an upgrade to a new minor release (3.2) it is recommended to check the [release notes](https://github.com/openssl/openssl/blob/openssl-3.2.0/NEWS.md#openssl-32) for the release. This lists the potentially breaking changes at the top.

**Reply by friesoft** [reply]
Thanks for the fast replies and the good news :) Yeah I've already checked out those pages 👍

**Comment by ware** [comment]
So happy to see this release.  Congratulations!

I don't see anything in the announcement or the [Release Strategy page](https://www.openssl.org/policies/releasestrat.html) on the intended support lifetime for 3.2.  I see the next feature release will be 3.3 in April but that seems orthogonal to how long 3.2 is intended to be supported.  Did I miss that somewhere?  Sorry if I missed something obvious.

**Reply by t8m** [reply]
Unless 3.2 is later declared to be an LTS release (which is unlikely) it will be supported for 2 years - i.e., the support ends on 23 Nov 2025.

**Reply by mattcaswell** [reply]
Hmm. We should update the release strategy page. The download page has already been updated with this information.

**Reply by mattcaswell** [reply]
https://github.com/openssl/web/pull/444

**Reply by ware** [reply]
Thank you folks.  That's what I **assumed** but didn't want to assume.

I should have thought about looking at the download page.  Doh!
